clippy_lints/src/format_args.rs

```diff
@@ -86,11 +149,54 @@ impl<'tcx> LateLintPass<'tcx> for FormatArgs {
                     check_format_in_format_args(cx, outermost_expn_data.call_site, name, arg.param.value);
                     check_to_string_in_format_args(cx, name, arg.param.value);
                 }
+
+                // if at least some of the arguments/format/precision are referenced by an index,
+                // e.g.  format!("{} {1}", foo, bar)  or  format!("{:1$}", foo, 2)
+                // we cannot remove an argument from a list until we support renumbering.
+                // We are OK if we inline all numbered arguments.
```

Comment:
nit: this comment should be a lil further down, it doesn't apply to the MSRV stuff